Src/AutoFixture.xUnit.net/FreezeAttribute.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Reflection;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.Xunit
+{
+    /// <summary>
+    /// An attribute that can be applied to parameters in an <see cref="AutoDataAttribute"/>-driven
+    /// Theory to indicate that the parameter value should be frozen and used to satisfy any requests
+    /// made through the same <see cref="IFixture"/> and that match a set of criteria.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
+    public sealed class FreezeAttribute : CustomizeAttribute
```

Comment:
My initial idea was have `[Freeze]` supersede the behavior of `[Frozen]`, which could then be made obsolete. However, I realized that the current set of matching possibilities don't include matching by a _specific type_ (like it's possible with the `[Frozen(As = typeof(Foo))]`) so they could probably be merged into the same attribute.

The question is whether the `As` and `By` matching behaviors should complement each other or be made mutually exclusive.

For example, it could be useful to be able to say:

```csharp
public void Test([Frozen(As = typeof(IFoo), By = Matching.PropertyName)]Bar p)
{
}
```

However, what happens if someone says:

```csharp
public void Test([Frozen(As = typeof(IFoo), By = Matching.ExactType)]Bar p)
{
}
```

Do we take into account either or both matching criteria? Does one override the other? How do we make it intuitive which behavior will apply?